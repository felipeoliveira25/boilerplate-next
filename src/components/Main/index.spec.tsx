import { render, screen } from '@testing-library/react'
import { Main } from '.'

describe('<Main />', () => {
    it('should render the heading', () => {
        //renderiza o componente
        const {container} = render(<Main title='Main para teste' description='teste main'/>)

        //busca o elemento e verifica a existência do mesmo
        expect(screen.getByRole('heading', { name: /main para teste/i })).toBeInTheDocument()

        //gerar snapshot para 'memorizar uma modelo que o componente deve seguir e teoricamente não deve ser alterado'
        expect(container.firstChild).toMatchSnapshot();
    })
})