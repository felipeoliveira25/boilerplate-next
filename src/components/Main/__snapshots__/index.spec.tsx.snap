// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the heading 1`] = `
.c0 {
  background-color: blue;
  width: 100%;
  height: 100%;
}

.c1 {
  color: #b0b0b0;
}

.c2 {
  color: #b2b2b2;
}

<main
  class="c0"
>
  <h1
    class="c1"
  >
    Main para teste
  </h1>
  <span
    class="c2"
  >
    teste main
  </span>
</main>
`;
